forc-plugins/forc-debug/src/server/handlers/handle_set_breakpoints.rs

```diff
@@ -0,0 +1,181 @@
+use crate::server::AdapterError;
+use crate::server::DapServer;
+use dap::requests::SetBreakpointsArguments;
+use dap::types::{Breakpoint, StartDebuggingRequestKind};
+use std::path::PathBuf;
+
+impl DapServer {
+    /// Handles a `set_breakpoints` request. Returns the list of [Breakpoint]s for the path provided in `args`.
+    pub(crate) fn handle_set_breakpoints(
+        &mut self,
+        args: &SetBreakpointsArguments,
+    ) -> Result<Vec<Breakpoint>, AdapterError> {
+        // Build the source maps so we can verify breakpoints
```

Comment:
> I am not sure about this, as a user I would not expect a breakpoint set to build anything.
It will only build once, and this will only happen after the user has started a debugging session. The user will only see the unverified breakpoints (greyed out) when in a debugging session.

> the breakpoints should be pending until it attaches to / launches the target.
This would work too, but I think the UX is a little better when building the source maps at this point, since they won't flash grey then red again.